prow/gerrit/reporter/reporter.go

```
@@ -103,46 +104,64 @@ func (c *Client) ShouldReport(pj *v1.ProwJob) bool {
+		logrus.Errorf("Gerrit report label not set for job %s", pj.Spec.Job)
+	} else {
+		selector[gerritReportLabel] = pj.ObjectMeta.Labels[gerritReportLabel]
+	}
+
+	// list all prowjobs in the patchset matching pj's type (pre- or post-submit)
+
+	pjsOnRevisionWithSameLabel, err := c.lister.List(selector.AsSelector())
 	if err != nil {
 		logrus.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
-		return err
+		return nil, err
 	}
 
 	// generate an aggregated report:
 	total := 0
 	success := 0
 	message := ""
-
-	for _, pjOnRevision := range pjsOnRevision {
-		if pjOnRevision.Status.PrevReportStates[c.GetName()] == pjOnRevision.Status.State {
-			logrus.Infof("Revision %s has been reported already", pj.ObjectMeta.Labels[clientGerritRevision])
-			return nil
+	var toReportJobs []*v1.ProwJob
+	mostRecentJob := map[string]*v1.ProwJob{}
+	for _, pjOnRevisionWithSameLabel := range pjsOnRevisionWithSameLabel {
+		job, ok := mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job]
+		if !ok || job.CreationTimestamp.Time.Before(pjOnRevisionWithSameLabel.CreationTimestamp.Time) {
+			mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job] = pjOnRevisionWithSameLabel
 		}
-
-		if pjOnRevision.Status.State == v1.AbortedState {
+	}
+	for _, pjOnRevisionWithSameLabel := range mostRecentJob {
+		if pjOnRevisionWithSameLabel.Status.State == v1.AbortedState {
 			continue
 		}
-
+		toReportJobs = append(toReportJobs, pjOnRevisionWithSameLabel)
 		total++
-		if pjOnRevision.Status.State == v1.SuccessState {
+		if pjOnRevisionWithSameLabel.Status.State == v1.SuccessState {
 			success++
 		}
 
-		message = fmt.Sprintf("%s\nJob %s finished with %s -- URL: %s", message, pjOnRevision.Spec.Job, pjOnRevision.Status.State, pjOnRevision.Status.URL)
+		message = fmt.Sprintf("%s\nJob %s finished with %s -- URL: %s", message, pjOnRevisionWithSameLabel.Spec.Job, pjOnRevisionWithSameLabel.Status.State, pjOnRevisionWithSameLabel.Status.URL)
+	}
+
+	if total <= 0 {
```

I think we should return an error? at least log something?

---

(Actually if this is an aborted job we are expecting total 0 here, maybe just log it)

---

done